### ISSUE TITLE
convert input_filter to python 3

### ISSUE THREAD

**Issue by madscatt** [madscatt/zazzie#162]
Noticed that the following python2 method crashes in python3, which breaks check_pdb_dcd()

`
def check_binary(filename):
    textchars = "".join(
        map(chr, [7, 8, 9, 10, 12, 13, 27] + range(0x20, 0x100)))

    def is_binary_string(bytes): return bool(bytes.translate(None, textchars))

    flag = is_binary_string(open(filename).read(1024))

    return flag

`

  File "/home/curtisj/anaconda3/lib/python3.9/site-packages/sassie/interface/input_filter.py", line 286, in check_binary
    map(chr, [7, 8, 9, 10, 12, 13, 27] + range(0x20, 0x100)))
TypeError: can only concatenate list (not "range") to list

**Comment by madscatt** [comment]
input_filter.py port status in this repo is solid and effectively complete for Python 3 at the module level.

Current status:

Primary file: src/sassie/interface/input_filter.py
Python 3 syntax check passed (compileall).
Dedicated short integration suite passed:
PYTHONPATH=src python -m pytest src/sassie/test_sassie/system_integration/short_tests/interface/input_filter/test_input_filter.py -q
Result: 63 passed
What I found:

No active Python 2 blockers in this file (xrange, py2 print, string.split, etc. are not in live code).
One legacy/comment-only py2 print string remains in a comment (non-functional).
Risk/notes around usage:

input_filter is a high-centrality dependency imported widely across tools/interface/simulate modules, so its current passing state is a strong base.
There are still some module files elsewhere using import input_filter (non-package-qualified) in specific interface modules (for example sascalc_filter.py, capriqorn_filter.py), but that is a caller-side import-style issue, not an input_filter.py port issue.
Bottom line:

src/sassie/interface/input_filter.py itself is Python 3-ready and test-backed.
Remaining risk is in downstream callers, not this module.
